### ISSUE TITLE
Shift-M and Shift-J does not find folders

### ISSUE THREAD

**Issue by FrankMittelbach** [RealRaven2000/QuickFolders#137]
I use the pro feature for moving mails a lot but it more or less stopped working for me. In the past you could start typing in a folder name and it would offer you possible completions. Now (since last release?) some folders are not found at all, so that it becomes impossible to move the mails. Even typing in the full folder name doesn't help.

If I put such "problem" folders into the quickfolder tab list then I can move mails using the mouse, but not via the keys.

I have unfortunately not found any pattern, some folders work others don't and those that don't sometimes start working again after a while.  For example I have a folder "aw" which wasn't found this morning at all. Now I just tried to look for "A..." because there is "Archive" and just typing in "A" suddenly showed me all folders including "aw" and now "a" also shows "aw" again ... so maybe some state issue with upper/lowercase?

**Comment by RealRaven2000** [comment]
> I use the pro feature for moving mails a lot but it more or less stopped working for me. In the past you could start typing in a folder name and it would offer you possible completions. Now (since last release?) some folders are not found at all, so that it becomes impossible to move the mails. Even typing in the full folder name doesn't help.
> 
> If I put such "problem" folders into the quickfolder tab list then I can move mails using the mouse, but not via the keys.
> 
> I have unfortunately not found any pattern, some folders work others don't and those that don't sometimes start working again after a while. For example I have a folder "aw" which wasn't found this morning at all. Now I just tried to look for "A..." because there is "Archive" and just typing in "A" suddenly showed me all folders including "aw" and now "a" also shows "aw" again ... so maybe some state issue with upper/lowercase?

There was a new feature for excluding accounts from the search results, check here first and make sure nothing is activated:

![image](https://user-images.githubusercontent.com/6314674/114390120-c1db4980-9b8d-11eb-9f80-beeffa2e57f9.png)

Here is also an option to only allow results from the same account, make sure that this isn't active.

You can also exclude single folders (and their child folders) using the "tab-specific properties":
![image](https://user-images.githubusercontent.com/6314674/114390341-08c93f00-9b8e-11eb-8654-ea9772a9c226.png)
![image](https://user-images.githubusercontent.com/6314674/114390399-1da5d280-9b8e-11eb-9bfc-e466b327f580.png)





As far as I know the folder search is not case sensitive, if you are getting reproducible results which differ based on what case you are entering a search I am very keen to find out)

**Comment by RealRaven2000** [comment]
> hi Axel,
> 
> nothing is activated (or deactivated) in the two menus. And yes, normally the shown results are not case sensitive.
> It is just that when typing "a" nothing showed nor when typing "aw" (the folder name). Then when I was half through typing in the bug report and tried "A" on a different mail suddenly all folders appeared. As I said, I haven't found a clear pattern, but the number of cases when Shift-M doesn't work was/is rather large after last update, but without a clear repeatable pattern (sometimes restarting thunderbird helps but not always either). So I'm grouping in the dark, but thought I report it.

Do your folder names contain spaces? I found one problem with a folder called Seamus R and it only brought up the QF tab in the search results until I entered the full first word "seamus " followed by the space. Definitely a bug / regression, possibly caused by changes to make quickfolders compatible with THunderbird 88 or by the changes to support excluding accounts... :(

**Comment by RealRaven2000** [comment]
> > hi Axel,
> > nothing is activated (or deactivated) in the two menus. And yes, normally the shown results are not case sensitive.
> > It is just that when typing "a" nothing showed nor when typing "aw" (the folder name). Then when I was half through typing in the bug report and tried "A" on a different mail suddenly all folders appeared. As I said, I haven't found a clear pattern, but the number of cases when Shift-M doesn't work was/is rather large after last update, but without a clear repeatable pattern (sometimes restarting thunderbird helps but not always either). So I'm grouping in the dark, but thought I report it.
> 
> Do your folder names contain spaces? I found one problem with a folder called Seamus R and it only brought up the QF tab in the search results until I entered the full first word "seamus " followed by the space. Definitely a bug / regression, possibly caused by changes to make quickfolders compatible with THunderbird 88 or by the changes to support excluding accounts... :(

no that one was a "false alarm" - the folder name only gets suppressed if it exists already in the search results (which it does because QuickFolders Tabs are searched first) - so if there is already a matching Tab, QuickFolders doesn't also display a duplicate of the folder itself...
![image](https://user-images.githubusercontent.com/6314674/114417343-6d45c780-9ba9-11eb-986a-451e3bda61a6.png)

the "little blue hat" already shows the tab pointing to the matching folder. So I still cannot reproduce the problem, so far...

**Comment by RealRaven2000** [comment]
> I use the pro feature for moving mails a lot but it more or less stopped working for me. In the past you could start typing in a folder name and it would offer you possible completions. Now (since last release?) some folders are not found at all, so that it becomes impossible to move the mails. Even typing in the full folder name doesn't help.
> 
> If I put such "problem" folders into the quickfolder tab list then I can move mails using the mouse, but not via the keys.
> 
> I have unfortunately not found any pattern, some folders work others don't and those that don't sometimes start working again after a while. For example I have a folder "aw" which wasn't found this morning at all. Now I just tried to look for "A..." because there is "Archive" and just typing in "A" suddenly showed me all folders including "aw" and now "a" also shows "aw" again ... so maybe some state issue with upper/lowercase?

Can you tell me what version of Thunderbird you are using? This is important because the code is "forked" and will do different things depending on whether you are using tb 68, 78 or 88.

**Comment by FrankMittelbach** [comment]
At the moment 78.9.1 

but back to the the problem behavior: I have 27 or so folders that contain an "a" and this morning nothing showed, now they all show yesterday it was not finding TLC3, right now everything is fine. If it appears again is there a way to do some debug setting?

**Comment by FrankMittelbach** [comment]
It seems to happen if I start from a special window, e.g., first show several mail in tree (Shift Cmd O) and then try to move one or several of them using Shift M

**Comment by FrankMittelbach** [comment]
maybe in that case it doesn't find any folder structure any more (due to the multiple account stuff added)

**Comment by fredstluka** [comment]
Frank,

Are you always using Shift-M and Shift-J from the same context
in Thunderbird?  I've reported a bug that they work differently
based on whether I'm looking at a single message, a list of
messages, a conversion thread, etc.  See:
- https://github.com/RealRaven2000/QuickFolders/issues/136
- 
https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-817934718

Maybe that explains some of the randomness you're seeing?

Just a thought...
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Frank/Axel,

Yes, this is exactly the same bug as #136:
- https://github.com/RealRaven2000/QuickFolders/issues/136

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by FrankMittelbach** [comment]
@fredstluka looks like like it, at least in the latest case where I could reproduce it.

**Comment by fredstluka** [comment]
Frank,

Good!  All the more reason for Axel to fix it soon.

BTW, did you see the workaround I posted at #136?

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
> Frank/Axel, Yes, this is exactly the same bug as #136: - #136
> […](#)

So what you are saying @FrankMittelbach - the emails are in a list that is not in the same folder? results of a search and clicking "open Email as list"? That area is not (well) tested or not tested at all. No guarantees!

**Comment by FrankMittelbach** [comment]
yes and no. I seldom use explicit searches and then display lists with "open Email as list", at least not when intending to use Shift-M. But one of my usual use cases is to do right click on an email  and then "In Themenbaum anzeigen" (Same as Shift-Cmd-O). The result is a search window I guess and the mails shown may or may not be in the same folder, e.g. some may already been archived --- anyway displaying mails by theme seems to me a natural and so I hope this is going to work again

**Comment by FrankMittelbach** [comment]
to give concrete usage example: our conversion here reaches me by email, so normally I would take one of the messages to show theme then deal with them and then select the all of them from that list and go shift-M to move them to some tmp folder ... but that stopped working.

I can move them invidiually from the inbox, yes, but they aren't consecutive so that is  a pain

**Comment by RealRaven2000** [comment]
> yes and no. I seldom use explicit searches and then display lists with "open Email as list", at least not when intending to use Shift-M. But one of my usual use cases is to do right click on an email and then "In Themenbaum anzeigen" (Same as Shift-Cmd-O). The result is a search window I guess and the mails shown may or may not be in the same folder, e.g. some may already been archived --- anyway displaying mails by theme seems to me a natural and so I hope this is going to work again

Ok - I usually move my emails (or have my filters do it) so the complete conversation is in the same folder. Since I also use "copy sent to current" my own replies will automatically be in the correct place. The Sent folder is a waste of time.

**Comment by RealRaven2000** [comment]
> Ok - I usually move my emails (or have my filters do it) so the complete conversation is in the same folder. Since I also use "copy sent to current" my own replies will automatically be in the correct place. The Sent folder is a waste of time.
> 
> Edit: that actually works so well (especially copy sent to current) on my production mail profile that it's actually hard for me to test that sort of thing. My threads are all in the same folder, pretty much always :) So I would have to cut them up and move them back to the inbox / sent folder or wherever they would be without all that.

This is seriously going to mess up my mails if I am not careful. I have my main mailbox both as a POP3 and an IMAP account, so  "open message in conversation" will actually display some mails duplicated. I need to be very careful that I don't move anything from IMAP. Not a fan of IMAP, it's dangerous and slow because of my bad internet connection, I usually leave stuff there well alone. If google chooses to nuke my account / mess up 12 years of mails I still have all my copies in my POP account.

**Comment by FrankMittelbach** [comment]
well workflow differs :-) 

I archive general mails by year but sometimes threads runs across several years (not uncomon in my world) so a look at a thread often contains mails from several folders (at least inbox + archive (+ possibly older  years))

so it is not a question of impa viz pop, the problem happens within a single account if you don't autosort with filters but only explicitly move stuff out of our inbox, but want to read stuff in your todo inbox in relation to stuff already handled. And this shift-M feature worked all the years I'm a pro user (and is more or less they only one I use from what pro offers)

**Comment by RealRaven2000** [comment]
> well workflow differs :-)
> 
> I archive general mails by year but sometimes threads runs across several years (not uncomon in my world) so a look at a thread often contains mails from several folders (at least inbox + archive (+ possibly older years))
> 
> so it is not a question of impa viz pop, the problem happens within a single account if you don't autosort with filters but only explicitly move stuff out of our inbox, but want to read stuff in your todo inbox in relation to stuff already handled. And this shift-M feature worked all the years I'm a pro user (and is more or less they only one I use from what pro offers)

I would love to know from which version onwards, can you disable automatic  updates and install older versions of QuickFolders from here?
https://addons.thunderbird.net/en-US/thunderbird/addon/quickfolders-tabbed-folders/versions/

And once it works again I can make a diff to see what changed.

**Comment by RealRaven2000** [comment]
...also have a look at JavaScript error console after it didn't work - any warnings / errors there?

**Comment by RealRaven2000** [comment]
Running one test, copying 2 Messages using Shift+T (same mechanism as moving via Shift+M) - one email was copied straight away, the other one is probably queued until I log of around midnight. See also my status bar:

![image](https://user-images.githubusercontent.com/6314674/114468248-10ff9980-9be3-11eb-881c-5325cc842260.png)

I really dislike IMAP. It's so unreliable, I don't know why people use it at all.

**Comment by FrankMittelbach** [comment]
> I would love to know from which version onwards, can you disable automatic updates and install older versions of QuickFolders from here?

sorry if this isa dumb question but where do you (un)set auto-update?

**Comment by FrankMittelbach** [comment]
ok found it. 

If I didn't make a mistake when trying the older versions then the last version where it works in a search window is 5.3 with my current installation. But I find that a bit surprising as I think it did work during March. Could it be that there is some dependency on firefox updates? Must confess that I don#t quite understand the dependencies between the two.

**Comment by FrankMittelbach** [comment]
> Running one test, copying 2 Messages using Shift+T (same mechanism as moving via Shift+M) - one email was copied straight away, the other one is probably queued until I log off around midnight and will never arrive in the folder. See also my status bar:
> 
> ![image](https://user-images.githubusercontent.com/6314674/114468248-10ff9980-9be3-11eb-881c-5325cc842260.png)
> 
> I really dislike IMAP. It's so unreliable, I don't know why people use it at all.

not quite clear why you think this has to do with IMAP or not. The point @fredstluka and I are making is that inside a window generated by Shift-Cmd-O a shift-M puts the cursor to the folder name box but if you start typing in characters there it doesn't display any matching folders (it finds none). I can't see how that relates to IMAP as the folder structure is local (and works (or worked) even if I'm offline without any connection to the IMAP server. I will check it with 5.3 later (currently have a big timemachine job running over the network so can#t take this machine here offline.

**Comment by RealRaven2000** [comment]
> > Running one test, copying 2 Messages using Shift+T (same mechanism as moving via Shift+M) - one email was copied straight away, the other one is probably queued until I log off around midnight and will never arrive in the folder. See also my status bar:
> > ![image](https://user-images.githubusercontent.com/6314674/114468248-10ff9980-9be3-11eb-881c-5325cc842260.png)
> > I really dislike IMAP. It's so unreliable, I don't know why people use it at all.
> 
> not quite clear why you think this has to do with IMAP or not. 

In that comment, I got confused with issue #135 (looking at search results from a email list)

> The point @fredstluka and I are making is that inside a window generated by Shift-Cmd-O a shift-M puts the cursor to the folder name box but if you start typing in characters there it doesn't display any matching folders (it finds none). 

"window" and result of a serach list - so that's important and wasn't listed in the original bug description or the title. What you are saying is that you open a new search list in a window and then the folder search doesn't work? The shortcuts Shift+Ctrl+O and Shift+Alt+O do not work here in Windows, so I don't know how to open this in a window? I have only tested the search field from the main (3pane) window.

**Comment by RealRaven2000** [comment]
If I select a **single message** and then use the command "open message in conversation" this will display a new tab - it's essentially a 3pane view without the folder pane (but that can be added again using view / layout / "folder pane"). In this view, folder search works for me too:

![image](https://user-images.githubusercontent.com/6314674/114540338-4d67df80-9c4d-11eb-8419-c41a07a799a0.png)

how do I force it to open in a new Window?

**Comment by RealRaven2000** [comment]
ok, so this only affects Thunderbird 78 and it's a side effect of the new "restrict search results to current / certain server(s). Since this bug is a duplicate - there is a solution here #136  - closing this one as duplicate!

**Comment by FrankMittelbach** [comment]
> "window" and result of a serach list - so that's important and wasn't listed in the original bug description or the title. What you are saying is that you open a new search list in a window and then the folder search doesn't work? The shortcuts Shift+Ctrl+O and Shift+Alt+O do not work here in Windows, so I don't know how to open this in a window? I have only tested the search field from the main (3pane) window.

sorry I was imprecise, I meant tab not window. 

But shift-strng-o works on windows I just checked. It appears to be greyed out if the mail has no theme siblings but if you have one with siblings then it exists and can be used (on the mac it works always).

As I don't use windows any longer I have only a very old machine which still has a thunderbird on it (explains why the quickfolder licence has expired I haven't used it since the start of this year. Starting it it updated quickfolder and here are some screen shots that show the problem:

1) a random mail in strng-shift-o view:

![ex1](https://user-images.githubusercontent.com/1062981/114547146-5ca35900-9c5e-11eb-8a0c-c3936af24af7.png)

2) now try adding "aw" to the folder box (there cursor is inside but not visible because of the screenshot tool--- no folders appear for completion):

![ex2](https://user-images.githubusercontent.com/1062981/114547334-9b391380-9c5e-11eb-8cba-280ca2bb7005.png)

3) now the same but in the inbox, now we get completions as expected:

![ex3](https://user-images.githubusercontent.com/1062981/114547414-b9067880-9c5e-11eb-88ea-ef0643edc7c2.png)

sorry, my windows is so slow that it is a pain to run thunderbird on it --- but it shows the same problem as on the mac (license or not)

**Comment by FrankMittelbach** [comment]
ah my painful windows attempt went on in parallel to the fix provided elsewhere... yes that fix seems to work, thanks.

**Comment by RealRaven2000** [comment]
> 3. now the same but in the inbox, now we get completions as expected:
> 
> ![ex3](https://user-images.githubusercontent.com/1062981/114547414-b9067880-9c5e-11eb-88ea-ef0643edc7c2.png)
> 

Unrelated, but...

your license is expired ;)

**Comment by FrankMittelbach** [comment]
> Unrelated, but...
> 
> your license is expired ;)

no it is not :-) ... it it only not installed on that windows machine (read the text above). I don't use the machine ... thnderbird takes 10 minutes to start (there imap is an issue I agree) so last time I used that machine was in 2020 and I never got around installing my license on that machine (and didn't for the screenshots either as they show the problem also without).

**Comment by RealRaven2000** [comment]
> > Unrelated, but...
> > your license is expired ;)
> 
> no it is not :-) ... it it only not installed on that windows machine (read the text above). I don't use the machine ... thnderbird takes 10 minutes to start (there imap is an issue I agree) so last time I used that machine was in 2020 and I never got around installing my license on that machine (and didn't for the screenshots either as they show the problem also without).

ok, no problem. Just thought I give  you a hint but I am not judging either way. :) X-D

**Comment by fredstluka** [comment]
Axel,

I agree about IMAP.  That's exactly the reason I always use POP
instead -- I don't want an error on a mail server to reach down
to my local box and delete decades of email.

I like the idea of keeping entire conversations in a single folder
via the TB "copy sent to current" option.  I never understood
how that was useful until you explained your scenario.  But, I
still prefer to have a single Sent folder that I can look in to
see a complete picture of my email sends on a given day.

I have TB set to BCC me on each message I send.  I take the
arrival of my copy as confirmation that my mail server is
working properly, meaning that the other recipients probably
got their copies.  That's useful since I DO run my own SMTP
server (but not my own POP server).

I manually file away (via QF -- thanks!) my BCC copy in the
folder with the rest of the conversation it belongs to, or just
delete it if it's not worth saving.  So, I end up with 2 copies of
each message I send.  One in Sent and another in a specific
folder (or in Trash).

Also, I use simple Delete for most emails, so they go to Trash,
but Shift-Delete for spam, junk, and emails sent to a list that
is archived well on the Web somewhere.  So those messages
don't go to Trash.  They simply get deleted.

I NEVER empty my Trash folder.  I just periodically move
messages from there to Trash_2020, Trash_2021, etc for
permanent archival.  So, even if I don't bother to file away a
message, I can (and often do) still find it years or decades
later.  That's where the Conversation view (Cmd-Shift-O)
comes in so handy.  it shows a single conversation that may
have crossed a year boundary, many still be ongoing, may
have been partially filed away, etc.

I could probably achieve the same effect via the native Archive
command in TB.  But I started my own convention way back
on Netscape Messenger or some earlier email client, long
before TB existed.  No good reason to change now.

I've also set the TB option to include the Trash folder in all
TB searches.  By default it used to be skipped.  Here are my
notes on how:
         - Right-click Trash | Properties
           | yes = Include messages in this folder in Global Search results

I've now read through all of the comments on this ticket and
agree that it is now completely fixed.

Thanks!
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------
